api/src/main/kotlin/com/dexportalapi/DeXPortalAPI/web/MainController.kt

```diff
@@ -0,0 +1,127 @@
+package com.dexportalapi.DeXPortalAPI
+
+import java.time.LocalDate
+import java.time.ZoneOffset
+import java.time.format.DateTimeFormatter
+import org.springframework.http.HttpStatus
+import org.springframework.http.MediaType.APPLICATION_FORM_URLENCODED
+import org.springframework.web.bind.annotation.CrossOrigin
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestHeader
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+import org.springframework.web.reactive.function.BodyInserters
+import org.springframework.web.reactive.function.client.*
+import org.springframework.web.server.ResponseStatusException
+
+/**
+ * TODO:
+ * - add CustomInternalServerErrorException
+ * - add CustomBadRequestException
```

Comment:
Did you still want to do this?

Reply:
I think it would be a good idea to in the future.